File: backend/models/posts.py
from datetime import datetime, timezone


def create_post_document(post):
    now = datetime.now(timezone.utc)
    comments = [
        {
            "author_name": comment.author_name,
            "content": comment.content,
            "created_at": now,
        }
        for comment in post.comments
    ]

    return {
        "title": post.title,
        "slug": post.slug,
        "content": post.content,
        "excerpt": post.excerpt,
        "author_name": post.author_name,
        "tags": post.tags,
        "comments": comments,
        "status": post.status,
        "created_at": now,
        "updated_at": now,
        "published_at": now if post.status == "published" else None,
    }

File: frontend/src/types/posts.ts
export type PostStatus = 'draft' | 'published'

export type PostComment = {
  _id?: string
  author_name: string
  content: string
  created_at?: string
}

export type CommentCreateInput = {
  author_name: string
  content: string
}

export type Post = {
  _id: string
  title: string
  slug: string
  content: string
  excerpt?: string | null
  author_name: string
  tags: string[]
  comments?: Array<PostComment | string>
  status: PostStatus | string
  created_at?: string
  updated_at?: string
  published_at?: string | null
}

export type PostCreateInput = {
  title: string
  slug: string
  content: string
  excerpt?: string | null
  author_name: string
  tags: string[]
}

export type PostFilters = {
  search: string
}
